api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -136,9 +136,8 @@ private Expression bindLiteralOperation(BoundTerm<T> boundTerm) {
     Literal<T> lit = literal().to(boundTerm.type());
 
     if (lit == null) {
-      throw new ValidationException(String.format(
-          "Invalid value for conversion to type %s: %s (%s)",
-          boundTerm.type(), literal().value(), literal().value().getClass().getName()));
+      throw new ValidationException("Invalid value for conversion to type %s: %s (%s)",
+          boundTerm.type(), literal().value(), literal().value().getClass().getName());
```

Comment:
So looking at the definition of the `literal()` function in this class, it seems it's possible for it to return null.

I guess it's not a concern as we would get NPE on the above call at line 136 when trying to call `.to` if `literal()` returned `null` before even getting to this part that calls `literal().value()`, but something I thought I'd bring up. Perhaps something we might follow up on in another issue or possibly I just missed the workflow that makes `literal()`'s result non-null by the time this `bindLiteralOperration` is called.